packages/http-client-python/emitter/src/emitter.ts

```diff
@@ -78,50 +80,113 @@ export async function $onEmit(context: EmitContext<PythonEmitterOptions>) {
   const root = path.join(dirname(fileURLToPath(import.meta.url)), "..", "..");
   const outputDir = context.emitterOutputDir;
   const yamlMap = emitCodeModel(sdkContext);
-  addDefaultOptions(sdkContext);
   const yamlPath = await saveCodeModelAsYaml("python-yaml-path", yamlMap);
-  let venvPath = path.join(root, "venv");
-  if (fs.existsSync(path.join(venvPath, "bin"))) {
-    venvPath = path.join(venvPath, "bin", "python");
-  } else if (fs.existsSync(path.join(venvPath, "Scripts"))) {
-    venvPath = path.join(venvPath, "Scripts", "python.exe");
-  } else {
-    throw new Error("Virtual environment doesn't exist.");
-  }
-  const commandArgs = [
-    venvPath,
-    `${root}/eng/scripts/setup/run_tsp.py`,
-    `--output-folder=${outputDir}`,
-    `--cadl-file=${yamlPath}`,
-  ];
+  addDefaultOptions(sdkContext);
   const resolvedOptions = sdkContext.emitContext.options;
-  if (resolvedOptions["packaging-files-config"]) {
-    const keyValuePairs = Object.entries(resolvedOptions["packaging-files-config"]).map(
-      ([key, value]) => {
-        return `${key}:${value}`;
-      },
-    );
-    commandArgs.push(`--packaging-files-config='${keyValuePairs.join("|")}'`);
-    resolvedOptions["packaging-files-config"] = undefined;
-  }
-  if (
-    resolvedOptions["package-pprint-name"] !== undefined &&
-    !resolvedOptions["package-pprint-name"].startsWith('"')
-  ) {
-    resolvedOptions["package-pprint-name"] = `"${resolvedOptions["package-pprint-name"]}"`;
-  }
+  if (resolvedOptions["use-pyodide"]) {
+    const commandArgs: Record<string, string> = {};
```

Comment:
the command args should be same whatever use pyodide or not, it's better to consolidate the logic.